lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetMetadataUtils.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.writer;
+
+import org.apache.parquet.column.statistics.BinaryStatistics;
+import org.apache.parquet.format.Statistics;
+import org.apache.parquet.internal.column.columnindex.BinaryTruncator;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.ColumnOrder;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+// Based on org.apache.parquet.format.converter.ParquetMetadataConverter
+public class ParquetMetadataUtils
+{
+    private ParquetMetadataUtils() {}
+
+    public static <T extends Comparable<T>> Statistics toParquetStatistics(org.apache.parquet.column.statistics.Statistics<T> stats, int truncateLength)
+    {
+        Statistics formatStats = new Statistics();
+        // We skip the `withinLimit(stats, truncateLength)` check from equivalent parquet-mr code
+        // which avoids writing stats altogether if they're above 4K in size.
+        // This is to allow readers to take advantage of truncated stats for filtering rather than having no stats.
+        if (!stats.isEmpty()) {
```

Comment:
Main change is here, rest is copy paste of parquet-mr code

Reply:
nit: can you copy/paste code in one commit and modify in another? would help identify changes clearly

Reply:
done